src/Avalonia.Wayland/WaylandPlatformThreading.cs

```diff
@@ -0,0 +1,245 @@
+using System;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Diagnostics;
+using Avalonia.Platform;
+using Avalonia.Threading;
+using Avalonia.Wayland;
+
+
+namespace Avalonia.Wayland
+{
+    unsafe class WaylandPlatformThreading : IPlatformThreadingInterface
```

Comment:
Is it just a copy-paste from X11 platform? If it is, we should just extract a base class, IMO.

Reply:
Thanks for looking at this.
Yes I agree. 

The main difference is that the X11  version adds an extra fd to the epoll set for receiving XEvents.. but the wayland version doesn't need that.

Where should I put the base class in the avalonia repo?

Reply:
Avalonia.Linux

we can make then Avalonia.Linux.X11, and Avalonia.Linux.Wayland

its ok to move things around because we are making breaking changes at the moment.